process_report/config.py

```diff
@@ -0,0 +1,184 @@
+    UPLOAD_TO_S3: bool = True
+
+    # S3 file paths
+    PI_S3_FILEPATH: str = "PIs/PI.csv"
+    ALIAS_S3_FILEPATH: str = "PIs/alias.csv"
+    PREPAY_DEBITS_S3_FILEPATH: str = "Prepay/prepay_debits.csv"
+
+    # Invoice file names
+    NONBILLABLE_INVOICE_NAME: str = "nonbillable"
+    BILLABLE_INVOICE_NAME: str = "billable"
+    BU_INTERNAL_INVOICE_NAME: str = "BU_Internal"
+    NERC_TOTAL_INVOICE_NAME: str = "NERC"
+    LENOVO_INVOICE_NAME: str = "Lenovo"
+    OCP_TEST_INVOICE_NAME: str = "OCP_TEST"
+    PI_SPECIFIC_FOLDER_NAME: str = "pi_invoices"
+
+    # Input file paths
+    NONBILLABLE_PIS_FILEPATH: pydantic.FilePath = "pi.txt"
+    NONBILLABLE_PROJECTS_FILEPATH: pydantic.FilePath = "projects.txt"
+    NONBILLABLE_TIMED_PROECTS_FILEPATH: pydantic.FilePath = "timed_projects.txt"
+
+    PREPAY_PROJECTS_FILEPATH: pydantic.FilePath = "prepaid_projects.csv"
+    PREPAY_CREDITS_FILEPATH: pydantic.FilePath = "prepaid_credits.csv"
+    PREPAY_CONTACTS_FILEPATH: pydantic.FilePath = "prepaid_contacts.csv"
+
+    # Defaults to being fetched from S3
+    CSV_INVOICE_FILEPATH_LIST: list[pydantic.FilePath] | None = None
+    PREPAY_DEBITS_FILEPATH: pydantic.FilePath | None = None
+    OLD_PI_FILEPATH: pydantic.FilePath | None = None
+    ALIAS_FILEPATH: pydantic.FilePath | None = None
+
+    # Defaults to being fetched in ColdfrontFetchProcessor
+    COLDFRONT_API_FILEPATH: pydantic.FilePath | None = None
+
+    # Defaults to being fetched in `nerc_rates`
+    NEW_PI_CREDIT_AMOUNT: Decimal | None = None
+    LIMIT_NEW_PI_CREDIT_TO_PARTNERS: bool | None = None
+    BU_SUBSIDY_AMOUNT: Decimal | None = None
+    LENOVO_CHARGE_INFO: dict[str, Decimal] | None = None
+
+    model_config = pydantic.ConfigDict(extra="forbid")
+
+    @staticmethod
+    def fetch_s3_invoices(invoice_month):
+        """Fetches usage invoices from S3 given invoice month"""
+        s3_invoice_list = list()
+        invoice_bucket = util.get_invoice_bucket()
+        for obj in invoice_bucket.objects.filter(
+            Prefix=f"Invoices/{invoice_month}/Service Invoices/"
+        ):
+            local_name = obj.key.split("/")[-1]
+            s3_invoice_list.append(local_name)
+            invoice_bucket.download_file(obj.key, local_name)
+
+        return s3_invoice_list
+
+    def get_csv_invoice_filepaths(self) -> list[pydantic.FilePath]:
+        if not self.CSV_INVOICE_FILEPATH_LIST:
+            self.CSV_INVOICE_FILEPATH_LIST = self.fetch_s3_invoices(self.INVOICE_MONTH)
+        return self.CSV_INVOICE_FILEPATH_LIST
+
+    def get_prepay_debits_filepath(self) -> pydantic.FilePath:
+        if not self.PREPAY_DEBITS_FILEPATH:
+            self.PREPAY_DEBITS_FILEPATH = util.fetch_s3(self.PREPAY_DEBITS_S3_FILEPATH)
+        return self.PREPAY_DEBITS_FILEPATH
+
+    def get_old_pi_filepath(self) -> pydantic.FilePath:
+        if not self.OLD_PI_FILEPATH:
+            self.OLD_PI_FILEPATH = util.fetch_s3(self.PI_S3_FILEPATH)
```

Comment:
Should we have better error handling around s3 access (e.g., in the case of incorrect auth, connection timeouts, etc)?